### ISSUE TITLE
Incorrect unused warning when value is used for a comparison inside a function invocation

### ISSUE THREAD

**Issue by cfal** [rust-lang/rust#64037]
The following code generates an unused variable warning for x:

```
fn test(v: bool) {
  println!("value is {}", v);
}

fn main() {
  let mut x: Option<u32> = None;
  x = Some(5);
  test(x.unwrap() == 5);
}
```

```
 --> test.rs:6:11
  |
6 |   let mut x: Option<u32> = None;
  |           ^
  |
  = note: `#[warn(unused_assignments)]` on by default
  = help: maybe it is overwritten before being read?
```

**Comment by jonas-schievink** [comment]
This is not an unused variable warning, the actual warning text is

    warning: value assigned to `x` is never read

...which is correct, since the initially assigned `None` is immediately overwritten with `Some(5)`
